portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,42 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> comments = new ArrayList<>();;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jie Liang!</h1>");
+    String json = convertToJsonUsingGson(comments);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = getComment(request);
+    if(isValidComment(comment)) {
```

Comment:
What if the comment is not valid?

Reply:
Currently nothing is done. Server could send 400 bad request error? and client would have to handle the error accordingly.

Else, client side form validation can be done too.

Reply:
You can choose which one you prefer. Just asking how would you handle invalid comment.